crates/zizmor/src/audit/impostor_commit.rs

```diff
@@ -111,6 +121,98 @@ impl ImpostorCommit {
         );
         Ok(true)
     }
+
+    /// Get the latest tag for a repository, preferring semantic version tags
+    fn get_latest_tag(&self, uses: &RepositoryUses) -> Result<Option<String>> {
+        let tags = self.client.list_tags(&uses.owner, &uses.repo)?;
+
+        // Prefer semantic version tags (starting with 'v') over other tags
+        let semver_tags: Vec<_> = tags
+            .iter()
+            .filter(|tag| {
+                tag.name.starts_with('v')
+                    && tag.name.chars().nth(1).is_some_and(|c| c.is_ascii_digit())
+            })
+            .collect();
+
+        let latest_tag = if !semver_tags.is_empty() {
+            // Use the first semver tag (GitHub returns tags in chronological order, newest first)
+            semver_tags.first().map(|tag| &tag.name)
+        } else {
+            // Fallback to the first tag if no semver tags are found
+            tags.first().map(|tag| &tag.name)
+        };
+
+        Ok(latest_tag.cloned())
+    }
+
+    /// Create a fix for an impostor commit by replacing it with the latest tag
+    fn create_impostor_fix<'doc, T>(&self, uses: &RepositoryUses, step: &T) -> Option<Fix<'doc>>
+    where
+        T: StepCommon<'doc> + for<'a> Routable<'a, 'doc>,
+    {
+        self.create_fix_for_location(uses, step.location().key, step.route().with_key("uses"))
+    }
+
+    /// Create a fix for a reusable workflow job
+    fn create_reusable_fix<'doc>(
+        &self,
+        uses: &RepositoryUses,
+        job: &ReusableWorkflowCallJob<'doc>,
+    ) -> Option<Fix<'doc>> {
+        self.create_fix_for_location(
+            uses,
+            job.location().key,
+            job.location().route.with_key("uses"),
+        )
+    }
+
+    /// Create a fix for the given location parameters
+    fn create_fix_for_location<'doc>(
+        &self,
+        uses: &RepositoryUses,
+        key: &'doc InputKey,
+        route: yamlpath::Route<'doc>,
+    ) -> Option<Fix<'doc>> {
+        // Get the latest tag for this repository
+        let latest_tag = match self.get_latest_tag(uses) {
+            Ok(Some(tag)) => tag,
+            Ok(None) => {
+                tracing::warn!(
+                    "No tags found for {}/{}, cannot create fix",
+                    uses.owner,
+                    uses.repo
+                );
+                return None;
+            }
+            Err(e) => {
+                tracing::warn!(
+                    "Failed to get latest tag for {}/{}: {}",
+                    uses.owner,
+                    uses.repo,
+                    e
+                );
+                return None;
+            }
```

Comment:
I _think_ this should be a hard error rather than a warning, unless I'm missing something.